optimum/habana/transformers/modeling_attn_mask_utils.py

```diff
@@ -35,28 +35,35 @@ def _make_causal_mask(
         device: torch.device,
         past_key_values_length: int = 0,
         sliding_window: Optional[int] = None,
+        token_idx: Optional[int] = 0,
```

Comment:
```suggestion
        token_idx: Optional[int] = None,
```

@imangohari1 why not set to `None`. If you set it to `0` by default then line 43 is obsolete since `token_idx` is never `None`

Reply:
or maybe there is a rational that I d not understand

Reply:
@yafshar  token_idx coming here shouldn't be None or zero, so I think this is just symantics, but I let @schoi-habana  chime in since she did this work.

Reply:
@yafshar you are right. since there's a check for None, it should be initialized as None. i've changed it locally but haven't pushed it. with the None, line 108-109 becomes redundant

Reply:
@yafshar  fixed: https://github.com/huggingface/optimum-habana/pull/2210/commits/5afdc231000b7e23841ebab83cd09b607b7a19f9

we still need lines 108-109 since token_idx is an input to `self._make_causal_mask`